src/common.rs

```diff
@@ -4,12 +4,13 @@ use std::{
     collections::VecDeque,
     fmt::Debug,
     sync::{Arc, Mutex},
+    thread,
 };
 
 /// Generic mock implementation.
 ///
-/// ⚠️ **Do not use this directly as end user! This is only a building
-/// block for creating mocks.**
+/// ⚠️ **Do not use this directly as end user! This is only a building block
+/// for creating mocks.**
```

Comment:
Why is it `pub` if it isn't supposed to be used as and user?

Reply:
Because our mocks are aliases too concrete instances of `GenericMock<T>`, for example `pub type Mock = Generic<Transaction>`.

If we change the visibility to `pub(crate)` and try to compile:

```
error[E0446]: crate-private type `common::Generic<eh0::spi::Transaction>` in public interface
   --> src/eh0/spi.rs:120:1
    |
120 | pub type Mock = Generic<Transaction>;
    | ^^^^^^^^^^^^^ can't leak crate-private type
    |
   ::: src/common.rs:24:1
    |
24  | pub(crate) struct Generic<T: Clone + Debug + PartialEq> {
    | ------------------------------------------------------- `common::Generic<eh0::spi::Transaction>` declared as crate-private
```